src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1824,6 +1824,12 @@ private void EnsureNestedTypesAreLoaded()
             }
         }
 
+        private void SetExtensionGroupingTypeNestedTypes(ArrayBuilder<PENamedTypeSymbol> groupingNestedTypes)
+        {
+            var exchangeResult = Interlocked.CompareExchange(ref _lazyNestedTypes, GroupByName(groupingNestedTypes), null);
+            Debug.Assert(exchangeResult == null);
```

Comment:
Why is this safe to assume? Is it not possible that multiple threads may get here at the same time? Certainly, given that `EnsureNestedTypesAreLoaded` does not assume that it is only called once, I do not expect it is safe to assume here either.

Reply:
> Why is this safe to assume? 

It is safe to assume by design and the assert verifies that the design is not violated, This method is called on a symbol that is not part of a regular container/member hierarchy, no one can gets its instance by accident, only through ```ExtensionGroupingType``` API (added below) on an extension symbol. No one can get an extension symbol from which this symbol is accessible before this function is called, because it is called before the extensions are added to their contained type.